crates/task-impls/src/network.rs

```diff
@@ -19,31 +18,82 @@ use hotshot_types::{
     },
     vote::{HasViewNumber, Vote},
 };
-use snafu::Snafu;
-use std::sync::Arc;
 use tracing::error;
 use tracing::instrument;
 
-/// the type of network task
-#[derive(Clone, Copy, Debug)]
-pub enum NetworkTaskKind {
-    /// quorum: the normal "everyone" committee
-    Quorum,
-    /// da committee
-    Committee,
-    /// view sync
-    ViewSync,
-    /// vid
-    VID,
+/// quorum filter
+pub fn quorum_filter<TYPES: NodeType>(event: &HotShotEvent<TYPES>) -> bool {
+    !matches!(
+        event,
+        HotShotEvent::QuorumProposalSend(_, _)
+            | HotShotEvent::QuorumVoteSend(_)
+            | HotShotEvent::Shutdown
+            | HotShotEvent::DACSend(_, _)
+            | HotShotEvent::ViewChange(_)
+            | HotShotEvent::TimeoutVoteSend(_)
+    )
+}
+
+/// committee filter
+pub fn committee_filter<TYPES: NodeType>(event: &HotShotEvent<TYPES>) -> bool {
+    !matches!(
+        event,
+        HotShotEvent::DAProposalSend(_, _)
+            | HotShotEvent::DAVoteSend(_)
+            | HotShotEvent::Shutdown
+            | HotShotEvent::ViewChange(_)
+    )
 }
 
+/// vid filter
+pub fn vid_filter<TYPES: NodeType>(event: &HotShotEvent<TYPES>) -> bool {
+    !matches!(
+        event,
+        HotShotEvent::Shutdown | HotShotEvent::VidDisperseSend(_, _) | HotShotEvent::ViewChange(_)
+    )
+}
+
+/// view sync filter
+pub fn view_sync_filter<TYPES: NodeType>(event: &HotShotEvent<TYPES>) -> bool {
+    !matches!(
+        event,
+        HotShotEvent::ViewSyncPreCommitCertificate2Send(_, _)
+            | HotShotEvent::ViewSyncCommitCertificate2Send(_, _)
+            | HotShotEvent::ViewSyncFinalizeCertificate2Send(_, _)
+            | HotShotEvent::ViewSyncPreCommitVoteSend(_)
+            | HotShotEvent::ViewSyncCommitVoteSend(_)
+            | HotShotEvent::ViewSyncFinalizeVoteSend(_)
+            | HotShotEvent::Shutdown
+            | HotShotEvent::ViewChange(_)
+    )
+}
 /// the network message task state
+#[derive(Clone)]
 pub struct NetworkMessageTaskState<TYPES: NodeType> {
-    /// event stream (used for publishing)
-    pub event_stream: ChannelStream<HotShotEvent<TYPES>>,
+    /// Sender to send internal events this task generates to other tasks
+    pub event_stream: Sender<HotShotEvent<TYPES>>,
 }
 
-impl<TYPES: NodeType> TS for NetworkMessageTaskState<TYPES> {}
+impl<TYPES: NodeType> TaskState for NetworkMessageTaskState<TYPES> {
+    type Event = Vec<Message<TYPES>>;
+    type Result = ();
+
+    async fn handle_event(event: Self::Event, task: &mut Task<Self>) -> Option<()>
+    where
+        Self: Sized,
+    {
+        task.state_mut().handle_messages(event).await;
+        None
+    }
+
+    fn filter(&self, _event: &Self::Event) -> bool {
+        false
+    }
+
+    fn should_shutdown(_event: &Self::Event) -> bool {
+        false
```

Comment:
Oh it is because this won't get HotShotEvents, it gets messages from the network layer which won't ever be a shutdown message.  So basically it runs until the network task closes the channel